apps/api/src/domain/models/digital-text/commands/add-photograph-to-digital-text-page/photograph-added-to-digital-text-page.event-handler.spec.ts

```diff
@@ -0,0 +1,193 @@
+import { PersistenceModule } from '../../../../../persistence/persistence.module';
+import generateDatabaseNameForTestSuite from '../../../../../persistence/repositories/__tests__/generateDatabaseNameForTestSuite';
+import { DigitalTextViewModel } from '../../../../../queries/digital-text';
+import { TestEventStream } from '../../../../../test-data/events';
+import buildDummyUuid from '../../../__tests__/utilities/buildDummyUuid';
+import { PhotographCreated } from '../../../photograph';
+import { IPhotographQueryRepository } from '../../../photograph/queries';
+import { PhotographViewModel } from '../../../photograph/queries/photograph.view-model';
+import { ArangoPhotographQueryRepository } from '../../../photograph/repositories';
+import { ArangoDigitalTextQueryRepository } from '../../queries/arango-digital-text-query-repository';
+import {
+    DIGITAL_TEXT_QUERY_REPOSITORY_PROVIDER_TOKEN,
+    IDigitalTextQueryRepository,
+} from '../../queries/digital-text-query-repository.interface';
+import { ContentAddedToDigitalTextPage } from '../add-content-to-digital-text-page';
+import { PageAddedToDigitalText } from '../add-page-to-digital-text/page-added-to-digital-text.event';
+import { DigitalTextCreated } from '../digital-text-created.event';
+import { PhotographAddedToDigitalTextPage } from './photograph-added-to-digital-text-page.event';
+import { PhotographAddedToDigitalTextPageEventHandler } from './photograph-added-to-digital-text-page.event-handler';
+
+const digitalTextId = buildDummyUuid(12);
+
+const compositeId = {
+    type: AggregateType.digitalText,
+    id: digitalTextId,
+};
+
+const digitalTextTitle = 'The new book';
+
+const languageCodeForTitle = LanguageCode.English;
+
+const digitalTextCreated = new TestEventStream().andThen<DigitalTextCreated>({
+    type: 'DIGITAL_TEXT_CREATED',
+    payload: {
+        title: digitalTextTitle,
+        languageCodeForTitle,
+    },
+});
+
+const pageIdentifier = '20';
+
+const digitalTextPageAdded = digitalTextCreated
+    .andThen<PageAddedToDigitalText>({
+        type: 'PAGE_ADDED_TO_DIGITAL_TEXT',
+        payload: {
+            identifier: pageIdentifier,
+        },
+    })
+    .andThen<ContentAddedToDigitalTextPage>({
+        type: 'CONTENT_ADDED_TO_DIGITAL_TEXT_PAGE',
+        payload: {
+            pageIdentifier,
+            text: 'dummy page content',
+            languageCode: LanguageCode.English,
+        },
+    });
+
+const photographId = buildDummyUuid(2);
+
+const photographTitle = 'A photo of a dolphin';
+
+const photographCreated = new TestEventStream().andThen<PhotographCreated>({
+    type: 'PHOTOGRAPH_CREATED',
+    payload: {
+        title: photographTitle,
+        languageCodeForTitle,
+    },
+});
+
+const [photographCreationEvent] = photographCreated.as({
+    type: AggregateType.photograph,
+    id: photographId,
+}) as [PhotographCreated];
+
+const existingPhotographView = PhotographViewModel.fromPhotographCreated(photographCreationEvent);
```

Comment:
I'm not sure you even need the event history. You can use state based setup:
```ts
const existingPhotographView = buildTestInstance(PhotographViewModel,{...})
```
See some of Blake's recent event-handlers for this pattern.